### PR TITLE
Import maddison lab

### PR DESCRIPTION
**Pull request by Merfoo**

### CODE REVIEW

File: app/views/otus/batch_load/castor/_batch_load.html.erb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<div>
+  <h2> Castor batch load </h2>
```
Comment (mjy) [inline app/views/otus/batch_load/castor/_batch_load.html.erb]: I think it might be worth renaming this.  It's generic enough that it's not just Castor specific.  "OTU with identifier" or something?

File: app/views/sequences/batch_load/genbank/_batch_load.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="flexbox full_viewport_whitespace">
+    <div class="item item1 one_third_width">
+      <h3> Description and requirements </h3>
+
+      <p>Creates sequences and origin relationships and attaches at most 2 identifiers to each sequence</p>
+      <ul>
+        <li>A tab-delimited, UTF-8 compatible text file is required.</li>
+        <li>Column headers are "filename" and "file_content" (see <%= link_to('sample template', asset_path('batch_load_templates/sequences/sequences_genbank_batch_load.tab'), target: '_blank') -%>).</li>
+        <li>Multiple uploads of the same data will fail.</li>
+        <li>Assumes a namespace with name "DRMSequenceId" and shortname "DRMSEQID" exists</li>
```
Comment (mjy) [inline app/views/sequences/batch_load/genbank/_batch_load.html.erb]: Don't worry about it now, but ultimately it would be nice to make this a user-definable parameter submitted on load.

File: app/views/collection_objects/batch_load/castor/_batch_load.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+      <p>
+        Creates collection objects, extracts, and taxon determinations from a Castor to TaxonWorks exported collection objects file. Attaches at most 5 identifiers to each collection object and at most 1 identifier to each extract.</p>
+      <ul>
+        <li>A tab-delimited, UTF-8 compatible text file is required.</li>
+        <li>Column headers are here <%= link_to('sample template', asset_path('batch_load_templates/collection_objects/collection_objects_castor_batch_load.tab'), target: '_blank') -%>).</li>
+        <li>Multiple uploads of the same data will do something, not sure.</li>
+        <li>Only creates collection objects if the column value of "locality_code_prefix" of the current row is "DRM"</li>
+        <li>There is currently some|no size limit to the file being loaded but this may change in the future.</li>
+        <li>Assumes a namespace with name "DRMLabVoucher" and shortname "DRMLV" exists</li>
```
Comment (mjy) [inline app/views/collection_objects/batch_load/castor/_batch_load.html.erb]: See other comment about making this a user-selectable param in the future.

File: app/views/collection_objects/batch_load/castor/_batch_load.html.erb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <li>Assumes a namespace with name "DRMDNAVoucher" and shortname "DRMDNA" exists</li>
```
Comment (mjy) [inline app/views/collection_objects/batch_load/castor/_batch_load.html.erb]: User param candidate as well.

File: app/views/sequences/batch_file_load/genbank/_batch_file_load.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      <p>Creates sequences and origin relationships and attaches at most 2 identifiers to each sequence</p>
+      <ul>
+        <li>A tab-delimited, UTF-8 compatible text file is required.</li>
+        <li>Sample file (see <%= link_to('&iSEQID00000349_&vDRMDNA4144_&g28S_&pPUB003_&nLionepha_chintimini_OR_4144.fas', asset_path('batch_file_load_templates/sequences/sequences_genbank_batch_load.tab'), target: '_blank') -%>).</li>
+        <li>Multiple uploads of the same data will fail.</li>
+        <li>Assumes a namespace with name "DRMSequenceId" and shortname "DRMSEQID" exists</li>
+        <li>Assumes filename has information about sequence</li>
+        <li>There is currently some|no size limit to the file being loaded but this may change in the future.</li>
+        <li>Identifier Global::GenBankAccessionCode will be attached with the value of the voucher number if not blank to each sequence</li>
+        <li>Identifier with namespace DRMSequenceId will be attached with the value of the sequence id if not blank to each sequence</li>
```
Comment (mjy) [inline app/views/sequences/batch_file_load/genbank/_batch_file_load.html.erb]: User selectable namespace param candidate for future change.

File: app/views/sequences/batch_load/genbank/_batch_load.html.erb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        <li>There is currently some|no size limit to the file being loaded but this may change in the future.</li>
+        <li>Identifier Global::GenBankAccessionCode will be attached with the value of the voucher number if not blank to each sequence</li>
+        <li>Identifier with namespace DRMSequenceId will be attached with the value of the sequence id if not blank to each sequence</li>
```
Comment (mjy) [inline app/views/sequences/batch_load/genbank/_batch_load.html.erb]: User selectable param candidate.

File: app/views/taxon_names/batch_load/castor/_batch_load.html.erb [hunk 7]
```diff
@@ -0,0 +1,39 @@
+<div>
+  <h2> Castor batch load </h2>
```
Comment (mjy) [inline app/views/taxon_names/batch_load/castor/_batch_load.html.erb]: Very cool.  Might be a candidate for renaming to a more generalized name, but maybe we just know it as the "Castor" version.

File: lib/batch_load/import/collecting_events/castor_interpreter.rb [hunk 8]
```diff
@@ -0,0 +1,124 @@
+# TODO: THIS IS A GENERATED STUB, it does not function
```
Comment (mjy) [inline lib/batch_load/import/collecting_events/castor_interpreter.rb]: Same comments as before re TODO comments.

File: app/views/otus/batch_load/castor/_batch_load.html.erb (hunk 1 above)
Comment (Merfoo) [inline app/views/otus/batch_load/castor/_batch_load.html.erb]: Perhaps this could be completed in a separate pull request? There would need to be multiple changes in different files and file renaming that would need to be done. I could just change the title for now so on the outlook it looks like "OTU with identifier" as well.

Comment (mjy) [inline app/views/otus/batch_load/castor/_batch_load.html.erb]: Definitely.
